src/query_plan/fetch_dependency_graph.rs

```diff
@@ -330,14 +661,22 @@ impl FetchInputs {
             supergraph_schema,
         }
     }
+
+    fn add(&self, selection: &NormalizedSelectionSet) {
+        assert!(
+            selection.schema.ptr_eq(&self.supergraph_schema),
+            "Inputs selections must be based on the supergraph schema"
+        );
+        todo!()
+    }
```

Comment:
`FetchInputs::add` needs to be ported from JS `GroupInputs.add` in a follow-up PR